src/main/java/com/google/research/bleth/simulator/AbstractSimulation.java

```diff
@@ -94,10 +106,35 @@ void resolverEstimate() {
     void writeRoundState() { }
 
     /** Gather statistical data of the current round and update the aggregated simulation statistics based on all rounds. */
-    void updateSimulationStats() { }
+    void updateSimulationStats() {
+        Map<Beacon, Location> beaconsToEstimatedLocations = ((GlobalResolver) resolver).getBeaconsToEstimatedLocations();
```

Comment:
Are you sure the cast here will always be valid? For all subclasses of the `AbstractSimulation`?
If so, consider just changing the field type.
Alternatively, add the method to the `IGlobalResolver` interface.
Avoid casts where you cannot definitely prove safety - otherwise they may manifest as runtime crashes. (In Google3 code, we have static analyzers that enforce this...)